Navigation/commands.py

```diff
@@ -19,32 +18,23 @@ def distance(channel, arg, user):
                         message = args[0].upper() + ' to ' + args[1].upper() + ': ' + '{:,.2f}ly'.format(dist)
                     else:
                         message = 'Invalid arguments. {}'.format(DIST_USAGE)
-                elif not len(args) == 2:
-                    message = 'Invalid number of systems. {}'.format(DIST_USAGE)
-                else:
-                    message = 'Mentions are not a valid parameter.'
             else:
                 message = 'Invalid number of systems. {}'.format(DIST_USAGE)
 
-            return slackbot.post_message(channel, message)
+            return await bot.post_message(channel, message)
 
-        @slackbot.command('rangemap', help='Get a dotlan map showing all locations in jump range from a system. '
+        @bot.command('rangemap', help='Get a dotlan map showing all locations in jump range from a system. '
                                            '{}'.format(RANGE_USAGE))
-        def rangemap(channel, arg, user):
-            slackbot.set_typing(channel)
+        async def rangemap(channel, arg, user):
             if arg:
                 args = arg.split(' ', -1)
-                if len(args):
+                if len(args) == 3:
```

Comment:
I've added this to prevent an index out of range in `get_dotlan_map`, removed the elif and else clauses because the Else wouldn't actually be reachable and the  catch-all else has the same message.